src/Transport/TestTransport.php

```diff
@@ -228,7 +233,23 @@ public function get(): iterable
             return [];
         }
 
-        return [\array_shift(self::$queue[$this->name])];
+        if (!$this->clock) {
```

Comment:
I think a project which does not use some sort of clock (any service with `\Psr\Clock\ClockInterface` as autowiring alias would do the trick here) should not support the stamp. They would not have any solution to "sleep" the clock transparently, and some tests could take ages!